lib/matplotlib/artist.py

```diff
@@ -845,21 +845,46 @@ def update(self, props):
         """
         store = self.eventson
         self.eventson = False
-        changed = False
+        try:
+            ret = [self._update_property(k, v)
+                   for k, v in sorted(props.items(), reverse=True)]
+        finally:
+            self.eventson = store
 
-        for k, v in six.iteritems(props):
-            if k in ['axes']:
-                setattr(self, k, v)
-            else:
-                func = getattr(self, 'set_' + k, None)
-                if func is None or not six.callable(func):
-                    raise AttributeError('Unknown property %s' % k)
-                func(v)
-            changed = True
-        self.eventson = store
-        if changed:
+        if len(ret):
             self.pchanged()
             self.stale = True
+        return ret
+
+    def _update_property(self, k, v):
```

Comment:
Maybe this could be an embedded function in `update()`, since it isn't used from anywhere else?